pf/internal/schemashim/block_schema.go

```diff
@@ -70,10 +77,19 @@ func (s *blockSchema) Elem() interface{} {
 	var schema shim.Schema
 	switch tt := t.(type) {
 	case types.MapType:
+		if r, ok := retObj(tt.ElemType); ok {
+			return r
+		}
 		schema = newTypeSchema(tt.ElemType, s.block.NestedAttrs())
 	case types.ListType:
+		if r, ok := retObj(tt.ElemType); ok {
+			return r
+		}
```

Comment:
This is the change in behavior that fixed the problem.

Reply:
This is great. This is the right fix but I've simplified the function and added some targeted unit tests. There's an additional problem that set-nested blocks were still not working properly.  Consider: https://github.com/pulumi/pulumi-terraform-bridge/pull/1018